src/sentry/receivers/features.py

```diff
@@ -0,0 +1,228 @@
+from sentry.plugins.bases.notify import NotificationPlugin
+from sentry.receivers.rules import DEFAULT_RULE_LABEL, DEFAULT_RULE_DATA
+from sentry.signals import (
+    alert_rule_created,
+    event_processed,
+    first_event_received,
+    project_created,
+    member_joined,
+    plugin_enabled,
+    user_feedback_received,
+    api_called,
+    issue_assigned,
+    issue_resolved_in_release,
+    advanced_search,
+    save_search_created,
+    inbound_filter_toggled,
+    sso_enabled,
+    data_scrubber_enabled,
+)
+from sentry.utils.javascript import has_sourcemap
+
+DEFAULT_TAGS = frozenset(['sentry_version', 'environment', 'level', 'logger',
+        'browser', 'browser.name', 'device', 'os', 'os.name', 'device', 'device.name',
+        'app.device', 'url', 'server_name', 'sentry:user', 'sentry:release', 'transaction',
+        'react', 'device.family'])
+
+
+# First Event
+@first_event_received.connect(weak=False)
+def record_first_event(project, group, **kwargs):
+    FeatureAdoption.objects.record(
+        organization_id=project.organization_id,
+        feature_slug="first_event",
+        complete=True)
+
+
+@event_processed.connect(weak=False)
+def record_event_processed(project, group, event, **kwargs):
+    # Platform
+    if group.platform in manager.location_slugs('language'):
+        FeatureAdoption.objects.record(
+            organization_id=project.organization_id,
+            feature_slug=group.platform,
+            complete=True)
+
+        # Frameworks
+        if event.data.get('sdk'):
+            if len(event.data.get('sdk').get('name').split(':')) == 2:
+                framework = event.data.get('sdk').get('name').split(':')[1]
+
+                if framework in manager.integration_slugs(group.platform):
+                    FeatureAdoption.objects.record(
+                        organization_id=project.organization_id,
+                        feature_slug=framework,
+                        complete=True)
+
+    elif event.data.get(get_interface('csp')):
+        FeatureAdoption.objects.record(
+            organization_id=project.organization_id,
+            feature_slug="csp",
+            complete=True)
+
+    # First Event
+    FeatureAdoption.objects.record(
+        organization_id=project.organization_id,
+        feature_slug="first_event",
+        complete=True)
+
+    # Release Tracking
+    if event.get_tag('sentry:release'):
+        FeatureAdoption.objects.record(
+            organization_id=project.organization_id,
+            feature_slug="release_tracking",
+            complete=True)
+
+    # Environment Tracking
+    if event.get_tag('environment'):
+        FeatureAdoption.objects.record(
+            organization_id=project.organization_id,
+            feature_slug="environment_tracking",
+            complete=True)
+
+    # User Tracking
+    if event.data.get('sentry.interfaces.User') and event.data.get('sentry.interfaces.User')[:3] != "ip:":
```

Comment:
For on-boarding purposes, I really want them to tag with actual user id.

Thoughts on backfilling the `sentry_organizationonboardingtask` to reset all User Context tasks to incomplete?